xt_cvdata/transforms/transforms.py

```diff
@@ -1,12 +1,37 @@
-__all__ = [
-    'Relabel',
-    'ToLabel',
-    'Squeeze',
-    'OneHot'
-]
 
 import torch 
 import numpy as np
+from PIL import Image
+from torchvision.transforms import functional as F
+import random
+
+__all__ = ['Relabel', 'ToLabel', 'Squeeze', 'OneHot', 'MaskCompose', 
+    'RandomHorizontalFlipSeg', 'RandomResizedCropSeg'
+]
+
+class MaskCompose:
+    """ Custom Composer of transforms used for applying transforms to the img, mask, or both.
+
+    MaskCompose([(Resize(24, 24), 'both'), (ColorJitter(), 'img'), (lambda x: x.long(), 'mask')])
+    """
+    target_options = ['img', 'mask', 'both']
+    def __init__(self, transforms):
+        self.transforms = transforms
+
+        assert all([target in self.target_options for _, target in transforms])
+    
+    def __call__(self, img, mask):
+        for t, target in self.transforms:
+            if target == 'img':
+                img = t(img)
+            elif target == 'mask':
+                mask = t(mask)
+            else:
+                # Concat then transform then split
```

Comment:
depricated comment

Reply:
I think it would be cleaner to concat along the channel dimension, then you don't have to reimplement all the transforms. The only ones you'd have to reimplement from torchvision would be ColorJitter, GrayScale, and RandomGrayScale (which you could skip because they are hardly ever used).

Reply:
Actually, you wouldn't have to implement any, those three I list above would only ever be applied in "img" mode

Reply:
Hmm, I hadn't quite thought it through. Is the issue that these are PIL images?